File: app/src/main/java/com/khmelenko/lab/varis/network/retrofit/github/GitHubRestClient.java
package com.khmelenko.lab.varis.network.retrofit.github;


import com.khmelenko.lab.varis.common.Constants;

import retrofit2.Retrofit;


/**
 * GitHub related REST client
 *
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
public class GitHubRestClient {

    private static final String GITHUB_URL = Constants.GITHUB_URL;

    private final Retrofit mRetrofit;

    private GithubApiService mGithubApiService;

    public GitHubRestClient(Retrofit retrofit) {
        mRetrofit = retrofit;

        // rest adapter for github API service
        Retrofit newRetrofit = mRetrofit.newBuilder()
                .baseUrl(GITHUB_URL)
                .build();
        mGithubApiService = newRetrofit.create(GithubApiService.class);
    }

    /**
     * Gets Github API service
     *
     * @return Github API service
     */
    public GithubApiService getApiService() {
        return mGithubApiService;
    }

}

File: app-v3/src/main/java/com/khmelenko/lab/varis/network/retrofit/github/GitHubRestClient.kt
package com.khmelenko.lab.varis.network.retrofit.github


import com.khmelenko.lab.varis.common.Constants
import okhttp3.OkHttpClient
import retrofit2.Retrofit

private const val GITHUB_URL = Constants.GITHUB_URL

/**
 * GitHub related REST client
 *
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
class GitHubRestClient(retrofit: Retrofit, okHttpClient: OkHttpClient) {

    /**
     * Gets Github API service
     *
     * @return Github API service
     */
    val apiService: GithubApiService = retrofit.newBuilder()
            .baseUrl(GITHUB_URL)
            .client(okHttpClient)
            .build()
            .create(GithubApiService::class.java)
}
